### ISSUE TITLE
Validate only on a master node during DDP .fit()

### ISSUE THREAD

**Issue by bongoramondo** [PyTorchLightning/pytorch-lightning#3915]
I encountered a problem, which I do not know how to properly address in PyTorch Lightning. My problem is as follows.
I am working on a retrieval problem, where the main aim is to find the same item on different photos, therefore each item has its' unique ID assigned in the dataset. For this task, when using DDP mode, I need to assure that all samples assigned to the given item (ID) are on the same GPU. As the Lightning in DDP mode, replaces the sampler with its own version,  I want to create my own sampler to split the data correctly among the GPUs, but I need to know the 'RANK' of each of the sub-processes. How can I access that information from a trainer, to use it in my custom sampler?

My main problem is however, during validation stage. To get the correct retrieval metrics I need to assure that all validation data is used for the retrieval, so the whole dataset needs to be available to a single GPU. The first way to do it that comes to my mind is to run the evaluation only in the main process, on the main GPU, is it possible at all in Lightning? 
Maybe I should use a custom Callback for that? 
Below I attached a snippets from my Trainer class.
 
#### Code

    def validation_step(self, batch, batch_idx):
        x, class_labels = batch
        with torch.no_grad():
            emb = self.model(x)
        return {"emb": emb, "labels": class_labels}

    def validation_epoch_end(self, outputs):
        embeddings = torch.cat([x["emb"] for x in outputs]).detach()
        labels = torch.cat([x["labels"] for x in outputs]).cpu().detach().numpy()
        mAP_metric = mAP_metric(self.trainer.datamodule.num_classes)
        mAP = self.mAP_metric.compute(embeddings=embeddings, labels=labels)
